airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/manifest.yaml

```diff
@@ -1,168 +1,1204 @@
+            endDate: >-
+              "{{ format_datetime(today_with_timezone(stream_slice.extra_fields['timezone']).strftime('%Y%m%d'), '%Y%m%d', '%Y%m%d') }}"
+        partition_router:
+          type: SubstreamPartitionRouter
+          parent_stream_configs:
+            - type: ParentStreamConfig
+              parent_key: "profileId"
+              stream: "#/definitions/streams/profiles_filtered"
+              partition_field: "profileId"
+              extra_fields:
+                - ["timezone"]
+        record_selector:
+          type: RecordSelector
+          extractor:
+            type: DpathExtractor
+            field_path: ["reports"]
+        paginator:
+          type: DefaultPaginator
+          page_token_option:
+            type: RequestOption
+            inject_into: body_json
+            field_name: cursorId
+          page_size_option:
+            type: RequestOption
+            inject_into: body_json
+            field_name: count
+          pagination_strategy:
+            type: CursorPagination
+            page_size: 300
+            cursor_value: "{{ response.cursorId }}"
+
+streams:
+  - $ref: "#/definitions/streams/profiles"
+  - $ref: "#/definitions/streams/portfolios"
+
+  - $ref: "#/definitions/streams/sponsored_display_campaigns"
+  - $ref: "#/definitions/streams/sponsored_display_ad_groups"
+  - $ref: "#/definitions/streams/sponsored_display_product_ads"
+  - $ref: "#/definitions/streams/sponsored_display_targetings"
+  - $ref: "#/definitions/streams/sponsored_display_creatives"
+  - $ref: "#/definitions/streams/sponsored_display_budget_rules"
+
+  - $ref: "#/definitions/streams/sponsored_brands_keywords"
+  - $ref: "#/definitions/streams/sponsored_brands_campaigns"
+  - $ref: "#/definitions/streams/sponsored_brands_ad_groups"
+
+  - $ref: "#/definitions/streams/sponsored_product_campaigns"
+  - $ref: "#/definitions/streams/sponsored_product_ad_groups"
+  - $ref: "#/definitions/streams/sponsored_product_keywords"
+  - $ref: "#/definitions/streams/sponsored_product_negative_keywords"
+  - $ref: "#/definitions/streams/sponsored_product_campaign_negative_keywords"
+  - $ref: "#/definitions/streams/sponsored_product_ads"
+  - $ref: "#/definitions/streams/sponsored_product_targetings"
+
+  - $ref: "#/definitions/streams/sponsored_product_ad_group_bid_recommendations"
+  - $ref: "#/definitions/streams/sponsored_product_ad_group_suggested_keywords"
+
+  - $ref: "#/definitions/streams/attribution_report_products"
+  - $ref: "#/definitions/streams/attribution_report_performance_adgroup"
+  - $ref: "#/definitions/streams/attribution_report_performance_campaign"
+  - $ref: "#/definitions/streams/attribution_report_performance_creative"
+
+# TODO: change concurrency after migration of async based reports
+concurrency_level:
+  type: ConcurrencyLevel
+  default_concurrency: 1
```

Comment:
Setting concurrency to 1 might be dangerous memory wise in some cases. What this does is that it will allow for only one thread to run at the same time. However, streams are processed in two different threads: one that generates the partitions and another that emits record from those partitions. If we were to have a case where there are a lot of partitions (thing about IssueComments in source-jira for example), we would generate all those partitions without a reader thread to consume them which could lead to memory issues. 

So regarding source-amazon-ads, I would have two questions:
* Do we have cases where we have a lot of partitions? I know we use a list partition router on the config profiles which should be small but do we have other cases? 
* Why are we afraid of enabling concurrency?

Reply:
> Do we have cases where we have a lot of partitions? I know we use a list partition router on the config profiles which should be small but do we have other cases?

No, only profile stream is used as a `Parent`, usually up to 5 profiles.

> Why are we afraid of enabling concurrency?
Not at all, I'll set this to default value.